### PR TITLE
add types for babel-plugin-macros

### PR DESCRIPTION
**Pull request by billykwok**
## Overview
Add types for babel-plugin-macros 2.6+
[GitHub Repo (https://github.com/kentcdodds/babel-plugin-macros)](https://github.com/kentcdodds/babel-plugin-macros)

## Template

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/babel-plugin-macros/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Type definitions for babel-plugin-macros 2.6
+// Project: https://github.com/kentcdodds/babel-plugin-macros
+// Definitions by: Billy Kwok <https://github.com/billykwok>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.6
+import * as Babel from 'babel__core';
```
Comment (andrewbranch) [inline types/babel-plugin-macros/index.d.ts]: I think this needs to be `@babel/core` (you already have it setup in your path mapping 👍)

File: types/babel-plugin-macros/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Type definitions for babel-plugin-macros 2.6
+// Project: https://github.com/kentcdodds/babel-plugin-macros
+// Definitions by: Billy Kwok <https://github.com/billykwok>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.6
```
Comment (andrewbranch) [inline types/babel-plugin-macros/index.d.ts]: This is the _minimum_ TypeScript version required. You should set it to the lowest value possible.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: types/babel-plugin-macros/index.d.ts (hunk 1 above)
Comment (billykwok) [reply types/babel-plugin-macros/index.d.ts]: Thanks for your comment. Fixed.

File: types/babel-plugin-macros/index.d.ts (hunk 2 above)
Comment (billykwok) [reply types/babel-plugin-macros/index.d.ts]: Thanks for your comment. Fixed.

**Review by andrewbranch — APPROVED** [review]

**Pull request merged**